File: 7-reverse-integer/reverse-integer.java
class Solution {
    public int reverse(int x) {
        int rd,n=0;
        while(x!=0){
            rd=x%10;
            if (n > Integer.MAX_VALUE / 10 || (n == Integer.MAX_VALUE / 10 && rd > 7)) return 0;
            if (n < Integer.MIN_VALUE / 10 || (n == Integer.MIN_VALUE / 10 && rd < -8)) return 0;

            n=n*10+rd;
            x=x/10;
        }return n;
    }
}

File: 7-reverse-integer/reverse-integer.cpp
class Solution {
public:
    int reverse(int x) {
         long int rev=0;
      while(x!=0){
        int rm=x%10;
        if (rev > INT_MAX / 10 || (rev == INT_MAX / 10 && rm > 7)) return 0;
        if (rev < INT_MIN / 10 || (rev == INT_MIN / 10 && rm < -8)) return 0;
        rev=rev*10+rm;
        x=x/10;
      }  
      return rev;
    }
};